sdk/src/main/java/com/uid2/utils/KeyUtils.kt

```diff
@@ -0,0 +1,82 @@
+package com.uid2.utils
+
+import com.uid2.extensions.decodeBase64
+import org.json.JSONArray
+import java.security.KeyFactory
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.PublicKey
+import java.security.SecureRandom
+import java.security.spec.ECGenParameterSpec
+import java.security.spec.X509EncodedKeySpec
+import javax.crypto.KeyAgreement
+import javax.crypto.SecretKey
+import javax.crypto.spec.SecretKeySpec
+
+/**
+ * A class containing utility methods for generating Security based keys, e.g. [PublicKey], [KeyPair] and [SecretKey].
+ * This is useful to allow these methods to be easily mocked for unit tests.
+ */
+internal class KeyUtils {
```

Comment:
should this be an object instead? or do you see value in being able to have multiple KeyUtils instances?

Reply:
`KeyUtils` is a class so that I can mock, to avoid UTs calling crypto specific code.

Reply:
Addressed with commits shared by Nacho!